super-csv/src/test/java/org/supercsv/io/TokenizerTest.java

```diff
@@ -586,6 +587,178 @@ public void testQuotedSpaces() throws Exception {
 		assertEquals("   three   ", columns.get(2));
 		assertEquals(input, tokenizer.getUntokenizedRow());
 	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithDoubleQuoteEscape() throws Exception {
+		final String input = "one,\"two\",\"field with \"\" quote char\",four";
+		tokenizer = createTokenizer(input, NORMAL_PREFERENCE);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithBackslashEscape() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "one,two,\"field with \\\" quote char\",four";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an escape character following another escape char.
+	 */
+	@Test
+	public void testEscapedEscapeChar() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"field with \\\\ escape char\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("field with \\ escape char", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an even number of escape characters
+	 * char.
+	 */
+	@Test
+	public void testEvenSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\\\"";
```

Comment:
if any odd sequence will do, how about just using `\\\` - it is easier to spot its an odd number.

Also, what are the escaped `"` doing here? do they play a role? if not leave them out?